pkg/virt-controller/watch/migration.go

```diff
@@ -1687,3 +1736,29 @@ func (c *MigrationController) isMigrationPolicyMatched(vmi *virtv1.VirtualMachin
 	migrationPolicyName := vmi.Status.MigrationState.MigrationPolicyName
 	return migrationPolicyName != nil && *migrationPolicyName != ""
 }
+
+func (c *MigrationController) isMigrationAbortedBeforeHandoff(migrationKey string) bool {
+	c.abortedMigrationsBeforeHandoffLock.Lock()
+	defer c.abortedMigrationsBeforeHandoffLock.Unlock()
+
+	_, isAborted := c.abortedMigrationsBeforeHandoff[migrationKey]
+	return isAborted
+}
+
+func (c *MigrationController) markMigrationAbortedBeforeHandoff(migrationKey string) {
+	c.abortedMigrationsBeforeHandoffLock.Lock()
+	defer c.abortedMigrationsBeforeHandoffLock.Unlock()
+
+	c.abortedMigrationsBeforeHandoff[migrationKey] = struct{}{}
+}
+
+func (c *MigrationController) unmarkMigrationAbortedBeforeHandoff(migrationKey string) {
+	c.abortedMigrationsBeforeHandoffLock.Lock()
+	defer c.abortedMigrationsBeforeHandoffLock.Unlock()
+
+	delete(c.abortedMigrationsBeforeHandoff, migrationKey)
+}
+
+func getMigrationKey(migration *virtv1.VirtualMachineInstanceMigration) string {
+	return migration.Namespace + "/" + migration.Name
+}
```

Comment:
we already have a function for this `key := controller.MigrationKey(migration)`

Reply:
done